Automattic/jetpack
# Fix Brute force activation when WAF disabled

Fixes #
- Separates the Brute force protection module activation/deactivation process from the WAF module activation/deactivation hooks so the modules can be handled independently of one another.

## Proposed changes:

* Removes `Brute_Force_Protection::instance()->on_activation()` from `Waf_Initializer:on_activation`
* Removes `Brute_Force_Protection::instance()->on_deactivation()` from `Waf_Initializer:on_deactivation`
* Creates a separate activation/deactivation hook for Brute force protection that runs regardless of whether the current environment is supported
* Updates naming

### Other information:


## Jetpack product discussion


- p1683637954396459-slack-C04E0KTGW9Z
- 1201069996155224-as-1204562568568412

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






On Jurassic Ninja/Tube with a mocked unsupported environment:
* Checkout this branch and start your Jurassic Tube on a clean docker instance or start a fresh Jurassic Ninja instance
* Define the WAF killswitch `DISABLE_JETPACK_WAF` in your `wp-config.php`
* Enable Jetpack and connect your user
* Proceed to `https://example.jurassic.tube/wp-admin/admin.php?page=jetpack#/settings`
* Verify that the WAF module is missing and Brute force module settings are available and enabled
* Verify that Brute force is working by visiting `/wp-admin` in an incognito browser window and failing attempts until you are met with a math auth
* Return the Brute force mode settings and add your current IP address to the allow list input field and save
* Ensure the settings are saved without error
* Return to the incognito login and verify that you are no longer met with a math auth regardless of the number of failed attempts

Alternatively, on a new Pressable or Atomic site:
- Install Jetpack and verify that you are unable to successfully update your Brute force allow list from the Jetpack settings page
- Deactivate Jetpack
- Install the Jetpack Beta Tester plugin and apply this branch of the Jetpack plugin and connect your user
- Return to the Brute force settings and ensure you are now able to add and save IPs to the allow list
- Verify that Brute force is working with and without your current IP allowed

Ensure no regression in WAF or Brute force functionality in Jetpack or Protect
